_posts/2021-11-09-logicdrop-automating-quarkus-with-gitlab.adoc

```diff
@@ -0,0 +1,249 @@
+- https://quarkus.io/guides/opentracing[Quarkus SmallRye OpenTracing^] hooks into https://docs.gitlab.com/ee/operations/tracing.html[GitLab Tracing^]
+
+Using the Quarkus extensions in combination with GitLab we get a complete picture of our environments, from development to deployment and monitoring, making building microservices easier. 
+
+== Dreaming of Pipelines
+
+=== Developer Powered Deployments
+Our most important pipelines revolve around making the developer process more streamlined and easier to work with. Developing, debugging, and collaborating on multiple services needed to be as easy and painless as possible.
+
+Previously, deployments to higher environments were tedious but doable but developers were not fully involved in this process. 
+
+TIP: Since moving to Quarkus, developers are now responsible for developing _and_ deploying (limitations apply) their own services.
+
+We decided to use GitLab https://docs.gitlab.com/ee/ci/pipelines/merge_request_pipelines.html[merge requests pipelines^] as the catalyst to drive our process.
+
+image::gitlab/merge-request.png[Merge Request,align="center"]
+
+On every push, the merge request will:
+
+- Run all the unit tests
+- Run any integration tests (Mongo, AWS, Redis)
+- Run any E2E tests (other external services)
+- Generate code coverage and quality reports
+- Publish an interactive Swagger API
+
+*The most useful feature of the merge request is that we can use it to deploy any number of services, and interact with them, into isolated environments.*
+
+image::gitlab/develop-pipeline.png[Review Pipeline,align="center"]
+
+_Review_ services lets us go beyond automated testing. We can spot-test only what is being worked on or even run a full suite of services if needed (this is especially useful during a combined UX and multiple service feature).
+
+TIP: Lighter than a feather, Quarkus native images are a small fraction of the size compared to their pure Java counterparts. This allows us to deploy a full set of services in less space then would be needed to deploy the equivalent Spring-Boot services.
+
+=== Building Quarkus Services
+The downstream Quarkus pipeline is a specialized set of jobs that handle building, testing, and containerizing Quarkus services only.
+
+For each changed service we:
+
+- Build a native image or FastJar
+- Run any tests (including native ones if needed)
+- Generate the Kubernetes manifests
+- Build and deploy its container
+
+image::gitlab/develop-downstream.png[Service Downstream Service,align="center"]
+
+TIP: We _only_ push the container and upload the manifests to AWS S3. This lets us accumulate changes into a single package, composed of one or more services, that can be deployed. 
+
+=== Faster Faster Pipelines
+To speed up the pipelines, especially when building multiple native images, we rely on GitLab AutoScaling runners to run the jobs in parallel.  This allows us to build any number of services in a relatively constant time frame.
```

Comment:
```suggestion
To speed up the pipelines, especially when building multiple native executables, we rely on GitLab AutoScaling runners to run the jobs in parallel.  This allows us to build any number of services in a relatively constant time frame.
```
In general, in Quarkus documentation, we prefer the term `native executable` to `native image`. Could you search and replace if it makes sense to you?

Reply:
Did this across the board.  Thank you for clarifying that.